common/domain/handler_test.go

```diff
@@ -1261,3 +1262,1221 @@ func TestHandler_UpdateAsyncWorkflowConfiguraton(t *testing.T) {
 		})
 	}
 }
+
+func TestHandler_UpdateDomain(t *testing.T) {
+	ctx := context.Background()
+	maxLength := 1
+
+	testCases := []struct {
+		name      string
+		setupMock func(
+			domainManager *persistence.MockDomainManager,
+			updateRequest *types.UpdateDomainRequest,
+			archivalMetadata *archiver.MockArchivalMetadata,
+			timeSource clock.MockedTimeSource,
+			domainReplicator *MockReplicator,
+		)
+		request  *types.UpdateDomainRequest
+		response *types.UpdateDomainResponse
+		err      error
+	}{
+		{
+			name: "Success case - global domain force failover",
+			setupMock: func(domainManager *persistence.MockDomainManager, updateRequest *types.UpdateDomainRequest, archivalMetadata *archiver.MockArchivalMetadata, timeSource clock.MockedTimeSource, domainReplicator *MockReplicator) {
+				domainResponse := &persistence.GetDomainResponse{
+					ReplicationConfig: &persistence.DomainReplicationConfig{
+						ActiveClusterName: cluster.TestCurrentClusterName,
+						Clusters: []*persistence.ClusterReplicationConfig{
+							{ClusterName: cluster.TestCurrentClusterName}, {ClusterName: cluster.TestAlternativeClusterName}},
+					},
+					Config: &persistence.DomainConfig{
+						Retention:                1,
+						EmitMetric:               true,
+						HistoryArchivalStatus:    types.ArchivalStatusDisabled,
+						VisibilityArchivalStatus: types.ArchivalStatusDisabled,
+						BadBinaries:              types.BadBinaries{Binaries: map[string]*types.BadBinaryInfo{}},
+						IsolationGroups:          types.IsolationGroupConfiguration{},
+						AsyncWorkflowConfig:      types.AsyncWorkflowConfiguration{Enabled: true},
+					},
+					Info: &persistence.DomainInfo{
+						Name:   constants.TestDomainName,
+						ID:     constants.TestDomainID,
+						Status: persistence.DomainStatusRegistered,
+					},
+					IsGlobalDomain:  true,
+					LastUpdatedTime: timeSource.Now().UnixNano(),
+					FailoverVersion: cluster.TestCurrentClusterInitialFailoverVersion,
+				}
+				domainManager.EXPECT().GetMetadata(ctx).Return(&persistence.GetMetadataResponse{}, nil).Times(1)
+				domainManager.EXPECT().GetDomain(ctx, &persistence.GetDomainRequest{Name: updateRequest.GetName()}).
+					Return(domainResponse, nil).Times(1)
+				archivalConfig := archiver.NewArchivalConfig(
+					common.ArchivalDisabled,
+					dynamicconfig.GetStringPropertyFn(common.ArchivalDisabled),
+					false,
+					dynamicconfig.GetBoolPropertyFn(false),
+					common.ArchivalDisabled,
+					"")
+				archivalMetadata.On("GetHistoryConfig").Return(archivalConfig).Times(1)
+				archivalMetadata.On("GetVisibilityConfig").Return(archivalConfig).Times(1)
+				timeSource.Advance(time.Hour)
```

Comment:
Not sure I understand the motivation for advancing the timesource?

Reply:
This is because the `FailoverCooldown` that we have in the handler `config`. Since we mock the `timeSource`, the check on this [line](https://github.com/uber/cadence/blob/master/common/domain/handler.go#L525) will always evaluate to true. By advancing the time we guarantee that we have "waited" for the cooldown to expire, and the check will be `false`, not returning an error